### PR TITLE
Use fixed size tables when possible (Reupload)

### PR DESCRIPTION
**Pull request by MaxGraey**
Some wasm engines will be able (or already can) to optimize indirect calls when table has fixed size (initial == maximum).

See for more details: https://bugzilla.mozilla.org/show_bug.cgi?id=1743586#c4

Here recreated version from #2189 

- [x] I've read the contributing guidelines
- [x] I've added my name and email to the NOTICE file

### CODE REVIEW

File: src/compiler.ts [hunk 1]
```diff
@@ -725,7 +725,16 @@ export class Compiler extends DiagnosticEmitter {
...
       functionTableNames[i] = functionTable[i].internalName;
     }
-    module.addFunctionTable("0", tableBase + functionTable.length, Module.UNLIMITED_TABLE, functionTableNames, module.i32(tableBase));
+
+    var tableSize = tableBase + functionTable.length;
+    module.addFunctionTable(
+      "0",
+      tableSize,
+      // use fixed size for non-imported tables
+      options.importTable ? Module.UNLIMITED_TABLE : tableSize,
```
Comment (dcodeIO) [inline src/compiler.ts:734]: Is this always desirable when the table is exported? Seems that `table.grow` could be called on it externally for some reason. Eventually, we might also want to call `table.grow` internally, which we can intercept, but the external one is a little tricky.

Comment (MaxGraey) [reply src/compiler.ts:734]: I can prohibit the same for exported tables

Comment (MaxGraey) [reply src/compiler.ts:734]: But table.grow will not work. Perhaps we should add a flag that allows you to expand the table? This is a very rare case anyway

Comment (dcodeIO) [reply src/compiler.ts]: Good thing is that `--exportTable` is not enabled by default, so depending on whether the option is given, it always does what's least surprising.

**Comment by dcodeIO** [comment]
👍

**Pull request merged**